src/build/app/mod.rs

```diff
@@ -1481,40 +1470,160 @@ impl<'b> App<'b> {
                 .collect();
 
             if !args_missing_help.is_empty() {
-                let abort_message = format!("AppSettings::HelpRequired is enabled for the App {}, but at least one of its arguments does not have either `help` or `long_help` set. List of such arguments: {}", app.name, args_missing_help.join(", "));
-                panic!(abort_message);
+                panic!(format!(
+                    "AppSettings::HelpRequired is enabled for the App {}, but at least one of its arguments does not have either `help` or `long_help` set. List of such arguments: {}",
+                    self.name,
+                    args_missing_help.join(", ")
+                ));
             }
         }
-        for sub_app in &app.subcommands {
-            Self::_panic_on_missing_help(&sub_app, help_required_globally);
+
+        for sub_app in &self.subcommands {
+            sub_app._panic_on_missing_help(help_required_globally);
         }
     }
 
     // Perform some expensive assertions on the Parser itself
-    fn _app_debug_asserts(&self) -> bool {
-        debugln!("App::_app_debug_asserts;");
-        for name in self.args.args.iter().map(|x| x.id) {
-            if self.args.args.iter().filter(|x| x.id == name).count() > 1 {
-                panic!(format!(
-                    "Arg names must be unique, found '{}' more than once",
-                    name
-                ));
+    #[allow(clippy::cognitive_complexity)]
+    fn _debug_asserts(&self) -> bool {
```

Comment:
No need to return `bool` here. This function ether panics or always returns true as I see.

Reply:
Yes, but this functions is used inside `assert!` which means it needs to return a `bool`.

Reply:
Those answers can be removed as useless because if function *did* return, it returned true.

Reply:
That `debug_assert!` is a shorter way for saying execute this only with debug_assertions. Also, we might want to add proper asserts which return bool later on.

Reply:
Code tends to be read much more than written. I don't think that picking obscure ways to do something is a good idea. This can very well be done via `if cfg!(debug_assertions) { ... }` which is not very long.

> we might want to add proper asserts which return bool later on.

I really doubt that. I can't imagine a check that doesn't panic right away.